### ISSUE TITLE
Should throw on invaid `(a = 1) = t`

### ISSUE THREAD

**Issue by fisker** [eslint/espree#470]
This should be invalid.

```text
require("espree").parse('(a = 1) = 2', {sourceType: 'module', ecmaVersion: 2019}).body[0].expression
Node {
  type: 'AssignmentExpression',
  start: 0,
  end: 11,
  operator: '=',
  left: Node {
    type: 'AssignmentPattern',
    start: 1,
    end: 6,
    left: Node { type: 'Identifier', start: 1, end: 2, name: 'a' },
    right: Node { type: 'Literal', start: 5, end: 6, value: 1, raw: '1' }
  },
  right: Node { type: 'Literal', start: 10, end: 11, value: 2, raw: '2' }
}
```

**Comment by fisker** [comment]
Sorry, false alarm.

**Comment by aladdin-add** [comment]
is it happening in acorn.js ?

**Comment by fisker** [comment]
As I tested, yes, it happens in `acorn` too. I'll open a issue there.

**Comment by aladdin-add** [comment]
as it's also a bug in espree, I suggest to keep it open to track it.

we can close it when the issue was fixed in acorn, and it  espree update its dependency acorn.

**Comment by mdjermanovic** [comment]
I think this is the same as https://github.com/acornjs/acorn/issues/886 which has been fixed by https://github.com/acornjs/acorn/commit/1d5b531b9178aa6c84c346a8d7ed381c17e9a820 and released in Acorn 8.0.0.

We'll update Acorn dependency in the next Espree major version (#456).
